rest/models/financials.go

```diff
@@ -0,0 +1,132 @@
+
+	PeriodOfReportDateEQ  *Date `query:"period_of_report_date"`
+	PeriodOfReportDateLT  *Date `query:"period_of_report_date.lt"`
+	PeriodOfReportDateLTE *Date `query:"period_of_report_date.lte"`
+	PeriodOfReportDateGT  *Date `query:"period_of_report_date.gt"`
+	PeriodOfReportDateGTE *Date `query:"period_of_report_date.gte"`
+
+	Timeframe *Timeframe `query:"timeframe"`
+
+	IncludeSources *bool `query:"include_sources"`
+
+	Sort  *Sort  `query:"sort"`
+	Order *Order `query:"order"`
+	Limit *int   `query:"limit"`
+}
+
+func (p ListFinancialsParams) WithTicker(q string) *ListFinancialsParams {
+	p.Ticker = &q
+	return &p
+}
+
+func (p ListFinancialsParams) WithCIK(q string) *ListFinancialsParams {
+	p.CIK = &q
+	return &p
+}
+
+func (p ListFinancialsParams) WithCompanyName(c NameComparator, q string) *ListFinancialsParams {
+	if c == FULL {
+		p.CompanyNameFULL = &q
+	} else if c == SEARCH {
+		p.CompanyNameSEARCH = &q
+	}
+	return &p
+}
+
+func (p ListFinancialsParams) WithSIC(q string) *ListFinancialsParams {
+	p.SIC = &q
+	return &p
+}
+
+func (p ListFinancialsParams) WithFilingDate(c Comparator, q Date) *ListFinancialsParams {
+	if c == EQ {
+		p.FilingDateEQ = &q
+	} else if c == LT {
+		p.FilingDateLT = &q
+	} else if c == LTE {
+		p.FilingDateLTE = &q
+	} else if c == GT {
+		p.FilingDateGT = &q
+	} else if c == GTE {
+		p.FilingDateGTE = &q
+	}
+	return &p
+}
+
+func (p ListFinancialsParams) WithPeriodOfReportDate(c Comparator, q Date) *ListFinancialsParams {
+	if c == EQ {
+		p.PeriodOfReportDateEQ = &q
+	} else if c == LT {
+		p.PeriodOfReportDateLT = &q
+	} else if c == LTE {
+		p.PeriodOfReportDateLTE = &q
+	} else if c == GT {
+		p.PeriodOfReportDateGT = &q
+	} else if c == GTE {
+		p.PeriodOfReportDateGTE = &q
+	}
+	return &p
+}
+
+func (p ListFinancialsParams) WithTimeframe(q Timeframe) *ListFinancialsParams {
+	p.Timeframe = &q
+	return &p
+}
+
+func (p ListFinancialsParams) WithIncludeSources(q bool) *ListFinancialsParams {
+	p.IncludeSources = &q
+	return &p
+}
+
+func (p ListFinancialsParams) WithSort(q Sort) *ListFinancialsParams {
+	p.Sort = &q
+	return &p
+}
+
+func (p ListFinancialsParams) WithOrder(q Order) *ListFinancialsParams {
+	p.Order = &q
+	return &p
+}
+
+func (p ListFinancialsParams) WithLimit(q int) *ListFinancialsParams {
+	p.Limit = &q
+	return &p
+}
+
+// ListFinancialsResponse is the response returned by the ListFinancials method.
+type ListFinancialsResponse struct {
+	BaseResponse
+	Results []Financial `json:"results,omitempty"`
+}
+
+// Financial contains detailed information on a specified stock financial.
+type Financial struct {
+	CIK                 string                 `json:"cik,omitempty"`
+	CompanyName         string                 `json:"company_name,omitempty"`
+	EndDate             string                 `json:"end_date,omitempty"`
+	FilingDate          string                 `json:"filing_date,omitempty"`
+	Financials          map[string]interface{} `json:"financials,omitempty"`
```

Comment:
this type is likely going to be hard to use. we should define a struct instead of using `interface{}`